docs/input.rst

```diff
@@ -223,10 +199,414 @@ a brace character in the literal text, it can be escaped by doubling:
 ``{{`` and ``}}``. [Python_Format]_
 
 
+Attributes
+----------
+
+Annotations or attributes apply to specific arguments or results.
+They describe semantic behavior for an argument.
+An attribute may be set to true by listing its name or
+it may have a value in parens::
+
+    - decl: Class1()  +name(new)
+    - decl: void Sum(int len, int *values+dimension+intent(in))
+    - decl: const std::string getName() +len(30)
+
+Attributes may also be added external to *decl*::
+
+    - decl: void Sum(int len, int *values)
+      attrs:
+          values:
+              dimension: True
+              intent: in  
+    - decl: const std::string getName()
+      fattrs:
+          len: 30
+  
+
+allocatable
+^^^^^^^^^^^
+
+Sometimes it is more convient to have the wrapper allocate an
+``intent(out)`` array before passing it to the C++ function.  This can
+be accomplished by adding the *allocatable* attribute.  For example the
+C++ function ``cos_doubles`` takes the cosine of an ``intent(in)``
+argument and assigns it to an ``intent(out)`` argument::
+
+    void cos_doubles(double *in, double *out, int size)
+    {
+        for(int i = 0; i < size; i++) {
+            out[i] = in[i] * 2.;
+        }
+    }
+
+This is wrapped as::
+
+    decl: void cos_doubles(double * in     +intent(in)  +dimension(:),
+                           double * out    +intent(out) +allocatable(mold=in),
+                           int      sizein +implied(size(in)))
+
+The *mold* argument is similar to the *mold* argument in the Fortran
+``allocate`` statement, it will allocate ``out`` as the same shape as
+``in``.  Also notice the use of the *implied* attribute on the
+``size`` argument.  This argument is not added to the Fortran API
+since its value is *implied* to be the size of argument ``in``.
+``size`` is the Fortran intrinsic which returns the number of items
+allocated by its argument.
+
+The Fortran wrapper produced is::
+
+    subroutine cos_doubles(in, out)
+        use iso_c_binding, only : C_DOUBLE, C_INT
+        real(C_DOUBLE), intent(IN) :: in(:)
+        real(C_DOUBLE), intent(OUT), allocatable :: out(:)
+        integer(C_INT) :: sizein
+        allocate(out, mold=in)
+        sizein = size(in)
+        call c_cos_doubles(in, out, sizein)
+    end subroutine cos_doubles
+
+The mold argument was added to the Fortran 2008 standard.  If the
+option **F_standard** is not 2008 then the allocate statement will be::
+
+        allocate(out(lbound(in,1):ubound(in,1)))
+
+
+For Python, same as *dimension*.
```

Comment:
I'm not following this sentence.  Was it meant to be more fleshed out?
(The ``dimension`` section below didn't help me, but perhaps adding a link would help?)

Reply:
Not sure what it means either.  I added a more complete sentence but much of this still needs to be more explicit.